virtcontainers/types/sandbox.go

```diff
@@ -50,12 +48,6 @@ type State struct {
 	// CgroupPath is the cgroup hierarchy where sandbox's processes
 	// including the hypervisor are placed.
 	CgroupPath string `json:"cgroupPath,omitempty"`
-
-	// Resources contains the resources assigned to the container.
-	// When a container is created resources specified in the config json
-	// are used, those resources change when a container is updated but
-	// the config json is not updated.
-	Resources specs.LinuxResources `json:"resources,omitempty"`
```

Comment:
@WeiZhang555 I don't think you should remove this. Look at the comment here, it states that at first the container resources are going to be identical to the container config, but later if those container resources are updated, the container config will not be updated, and that's why this resource structure is important. It will hold the new values.

Reply:
Let me test it. I guess you're right. How about update it back to config?

Reply:
Commit is updated. @sboeuf 

The updated resource will be write back to config in https://github.com/kata-containers/runtime/pull/1416/files#diff-8a56f5a1fa5971cc43e650ffa24f1e2bR1362 .

The problem here is, we store container resources both in state file and config file, it's not necessary. I am trying to remove duplicated fields and only keep necessary fields.

If we want to original unchanged resources, we can still find it in Annotations[configJSONKey], right? So from every point, there's one unnecessary copy here.

Reply:
I think the bottom question is, do we expect applied container resources to be different from the expected container resources? For instance, could we decide to give a container more memory because we're rounding things up, but when you do the computation when a new container is created, it's actually better to sum up from the config than the state?
I'm trying to understand if we have added value from keeping config information.

Reply:
@sboeuf I think you're talking about another issue. Current implementation keeps 3 copy of resources:
1. `Annotations[configJSONKey]`: keeps original unchanged OCI spec with resources.
2. `config.json`: one copy of OCI spec resources without change.
3. `state.json`: resources container really get.

My idea is that we can merge 2 and 3. because we already have a copy of unchanged resources, we don't need a duplicated copy.

If the PR is removing `spec.Resources` from `config.json`, it may be easier to understand. But I prefer container resources stay in config because it sounds more reasonable.

Reply:
And actually we have fourth copy of `Resources`, it's from `bundle/config.json`

Reply:
That's sounds fair to me. As long as we always have a way to know about the initial resources of the container and to know about the current resources, this is fine.
And `config` sounds like a good candidate since `state` is something that is supposed to hold values that have been generated by our runtime. If we consider the container resources fully dictated by the configuration, I'm happy with having those information sitting in the stored container config.

Reply:
Thanks @sboeuf :)